### PR TITLE
Remove motor sequence number from AP_Motors structures

### PR DESCRIPTION
**Pull request by peterbarker**
.... it can be infered from the position in the structure.

### CODE REVIEW

**Comment by peterbarker** [comment]
One commit on top of https://github.com/ArduPilot/ardupilot/pull/18525

File: libraries/AP_Motors/AP_MotorsMatrix.cpp [hunk 1]
```diff
@@ -844,86 +859,109 @@ void AP_MotorsMatrix::setup_motors(motor_frame_class frame_class, motor_frame_ty
...
-                    add_motor_raw(AP_MOTORS_MOT_1, 0.333f, -1.0f, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  1);
-                    add_motor_raw(AP_MOTORS_MOT_2, -0.333f,  1.0f, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  5);
-                    add_motor_raw(AP_MOTORS_MOT_3,    1.0f, -1.0f, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 2);
-                    add_motor_raw(AP_MOTORS_MOT_4,  0.333f,  1.0f, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 4);
-                    add_motor_raw(AP_MOTORS_MOT_5, -0.333f, -1.0f, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 8);
-                    add_motor_raw(AP_MOTORS_MOT_6,   -1.0f,  1.0f, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 6);
-                    add_motor_raw(AP_MOTORS_MOT_7,   -1.0f, -1.0f, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  7);
-                    add_motor_raw(AP_MOTORS_MOT_8,    1.0f,  1.0f, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  3);
+                    static const AP_MotorsMatrix::MotorDefRaw motors[] {
+                        { 0.333f, -1.0f, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  1 },
```
Comment (amilcarlucas) [inline libraries/AP_Motors/AP_MotorsMatrix.cpp:923]: whitespace

File: libraries/AP_Motors/AP_MotorsMatrix.h [hunk 2]
```diff
@@ -90,14 +90,33 @@ class AP_MotorsMatrix : public AP_MotorsMulticopter {
...
+        uint8_t testing_order;
+    };
 
     // method to add many motors specified in a structure:
-    void add_motors(const struct MotorDefInt *motor, uint8_t num_motors);
+    template <typename T>
+    void add_motors(T *motor, uint8_t num_motors);
+
+    // structure used for initialising motors that add have separate
+    // roll/pitch/throttle factors.  Note that this does *not* include
```
Comment (amilcarlucas) [inline libraries/AP_Motors/AP_MotorsMatrix.h:109]:
```suggestion
    // roll/pitch/yaw throttle factors.  Note that this does *not* include
```

**Comment by amilcarlucas** [comment]
```
Binary           text              data         bss          total
---------------  ----------------  -----------  -----------  ----------------
ardusub          -2224 (-0.1453%)  0 (0.0000%)  0 (0.0000%)  -2224 (-0.1240%)
blimp            0 (0.0000%)       0 (0.0000%)  0 (0.0000%)  0 (0.0000%)
arducopter-heli  0 (0.0000%)       0 (0.0000%)  0 (0.0000%)  0 (0.0000%)
arducopter       -2224 (-0.1280%)  0 (0.0000%)  0 (0.0000%)  -2224 (-0.1112%)
ardurover        0 (0.0000%)       0 (0.0000%)  0 (0.0000%)  0 (0.0000%)
antennatracker   0 (0.0000%)       0 (0.0000%)  0 (0.0000%)  0 (0.0000%)
arduplane        -2248 (-0.1296%)  0 (0.0000%)  0 (0.0000%)  -2248 (-0.1126%)
```

File: libraries/AP_Motors/AP_MotorsMatrix.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_Motors/AP_MotorsMatrix.cpp:923]: Pre-existing; I'm doing this mechanically :-)

File: libraries/AP_Motors/AP_MotorsMatrix.h (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_Motors/AP_MotorsMatrix.h]: Fixed in PR this is based on, and this one rebased.

**Review by IamPete1 — APPROVED** [review]
Looks good, a nice flash saving.

**Pull request closed without merging**